### PR TITLE
RDAMT-281: Created Standalone console App for running locally. Added …

### PR DESCRIPTION
**Pull request by peterchafe**
…app Version to Page, and added custom title to image.

### CODE REVIEW

File: Web/js/app.js [hunk 1]
```diff
@@ -11,6 +11,18 @@ const configureClient = async () => {
...
+        if (config.version == null)
+        {
+            versionSpans[i].innerHTML = "";
+        }
+        else
+        {
+            versionSpans[i].innerHTML = config.version;
+        }
+    }
+
```
Comment (stephenmatis) [inline Web/js/app.js:18]: I think this could be shortened to:

`versionSpans[i].innerHTML = config.version || "";`

File: DotNet/RoadmapLogic/Input.cs [hunk 2]
```diff
@@ -7,17 +7,19 @@ namespace RoadmapLogic
     public class Input
     {
         public Input(
+            string title,
             string team,
-            string startDate,
+            string start_date,
```
Comment (stephenmatis) [inline DotNet/RoadmapLogic/Input.cs]: Why did you rename this arg? If it works as `startDate` (it did, right?) then I would leave it that way to match coding standards.

File: DotNet/RoadmapLogic/Input.cs [hunk 3]
```diff
@@ -7,17 +7,19 @@ namespace RoadmapLogic
     public class Input
     {
         public Input(
+            string title,
```
Comment (stephenmatis) [inline DotNet/RoadmapLogic/Input.cs]: The title addition is good, but I don't like seeing all the `null` arguments passed in. Could you make this an optional argument after `quarters` with a default value of `null`? Then all the calling code would not have to change.

**Review by stephenmatis — COMMENTED** [review]
I think you need a return if user runs .exe without supplying an argument.

image

Can you add a new section to the README showing how to use the RoadmapApp?

File: Web/js/app.js (hunk 1 above)
Comment (peterchafe) [reply Web/js/app.js:18]: Good catch, my js coding is quite limited. Thanks.

File: DotNet/RoadmapLogic/Input.cs (hunk 3 above)
Comment (peterchafe) [reply DotNet/RoadmapLogic/Input.cs]: I had been thinking about the same, not sure why I went with adding the first argument. this is much cleaner. Thanks.

File: DotNet/RoadmapLogic/Input.cs (hunk 2 above)
Comment (peterchafe) [reply DotNet/RoadmapLogic/Input.cs]: I am unsure why this is the case, but when it is set to the original, it is unable to parse the json file properly.
When the start_date variable matches that of the json file, it parses fine.

Perhaps you could test it out, and see that you get the same result.

Thanks.

Comment (stephenmatis) [reply DotNet/RoadmapLogic/Input.cs]: Changed back to startDate and fixed by adding a Newtonsoft property to the argument instead.

**Review by stephenmatis — APPROVED** [review]
